src/methods/BaseCompare.js

```diff
@@ -126,12 +126,13 @@ export default class BaseCompare {
     return path.join(runtimeConfigPath, `${name}-${suffix}.json`);
   }
 
-  createResultReport(misMatchPercentage, isWithinMisMatchTolerance, isSameDimensions) {
+  createResultReport(misMatchPercentage, isWithinMisMatchTolerance, isSameDimensions, filePaths) {
```

Comment:
> Moreover this is to inflexibel and only useful for file comparison. I suggest to replace filePaths with an object that wil be assigned via Object.assign to the returned object. Then custom or other compare methods can enhance the report with additional information.

This method should be more generic like I already said. What I mean is the following:

```js
  createResultReport(misMatchPercentage, isWithinMisMatchTolerance, isSameDimensions, options = {}) {
    return {
      ...options,
      misMatchPercentage,
      isWithinMisMatchTolerance,
      isSameDimensions,
      isExactSameImage: misMatchPercentage === 0,
    };
  }
````
This allows you to pass everything you want in and just merges this into the report object.

In your case you would call it the following for LocalCompare & SaveScreenshot. All other methods shouldn't be touched.

```js
this.createResultReport(misMatchPercentage, true, isSameDimensions, {
  filenames: {
    reference: referencePath,
    screenshot: screenshotPath,
    diff: null,
  },
});
```

Reply:
OK. Thanks for that. Had to look into the spread operator in ES6 because I thought ...options was a typo. XD